tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,554 @@
+    expect(root.find(Notice)).toHaveLength(0);
+
+    expect(root.find('.UserProfileEdit--Card').first())
+      .toHaveProp('header', 'Account');
+    expect(root.find('.UserProfileEdit-email--help')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-aside')).toHaveText(oneLine`Tell users a
+      bit more information about yourself. These fields are optional, but
+      they'll help other users get to know you better.`
+    );
+    expect(root.find({ htmlFor: 'biography' }))
+      .toHaveText('Introduce yourself to the community if you like');
+  });
+
+  it('dispatches fetchUserAccount action if username is not found', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const username = 'i-am-not-tofumatt';
+    const root = renderUserProfileEdit({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount action if there is no username', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    // This happens when loading the user edit profile page of the current
+    // logged user (e.g., page refresh).
+    renderUserProfileEdit({ params: {}, store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('dispatches fetchUserAccount action if username changes', () => {
+    const username = 'black-panther';
+    const params = { username };
+
+    const { store } = signInUserWithUsername(username);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfileEdit({ params, store });
+
+    dispatchSpy.reset();
+
+    // We set `user` to `null` because that's what `mapStateToProps()` would do
+    // because the user is not loaded yet.
+    root.setProps({ username: 'killmonger', user: null });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username: 'killmonger',
+    }));
+
+    expect(root.find('.UserProfileEdit-location')).toHaveProp('value', '');
+  });
+
+  it('does not dispatch fetchUserAccount if username does not change', () => {
+    const username = 'black-panther';
+    const params = { username };
+
+    const { store } = signInUserWithUsername(username);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfileEdit({ params, store });
+
+    dispatchSpy.reset();
+
+    root.setProps({ params });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders a username input field', () => {
+    const root = renderUserProfileEdit();
+    expect(root.find('.UserProfileEdit-username')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-username'))
+      .toHaveProp('value', defaultUserProps.username);
```

Comment:
No you're 200% right, we should be doing this and I am daft. I knew this but wanted to write more test cases and tried to save time here...

Reply:
Well, like I said, it would be my inclination to write it this way as well, so you're calling both of us daft. ;-)  But, yeah, we sometimes need to write more code than we'd like to. :-)